### ISSUE TITLE
Add addToTransactions to Attachment

### ISSUE THREAD

**Issue by philipphofmann** [getsentry/sentry-cocoa#1606]
### Problem Statement

The Attachment doesn't have the property `addToTransactions`. [Java](https://github.com/getsentry/sentry-java/blob/9f87477ac412f3677df87304877ce598b14d7f2c/sentry/src/main/java/io/sentry/Attachment.java#L14) and [Python](https://github.com/getsentry/sentry-python/blob/e971cafb896aa9bef0fdfb8df2588d42752aad4b/sentry_sdk/attachments.py#L18) have it.

Currently, the attachments get attached to every transaction, but the default of `addToTransactions` is `false`.

### Solution Brainstorm

Add `addToTransactions` to `SentryAttachment` and keep the default value of `true`. Change the default value to `false` in the next major release to align with other SDKs. If we wait until 8.0.0 to do this, we can already change the default value to `true`.

**Comment by philipphofmann** [comment]
It's worth noting that attachments for transactions currently don't work on Cocoa, see https://getsentry.atlassian.net/browse/INGEST-815.

**Comment by philipphofmann** [comment]
Please also update the user-facing docs https://docs.sentry.io/platforms/apple/enriching-events/attachments/.

**Comment by brustolin** [comment]
Now that we're creating 8.0, should `attachToTransaction` be false by default?

**Comment by brustolin** [comment]
I believe this is still blocked by https://getsentry.atlassian.net/browse/INGEST-815 right?

**Comment by philipphofmann** [comment]
> I believe this is still blocked by https://getsentry.atlassian.net/browse/INGEST-815 right?

That's news to me. If that's true, then yes, it doesn't make sense to add it right now. We could clarify what the expected timeline for that is.

**Comment by brustolin** [comment]
This is still blocked and  Ingest has no ETA for it

**Comment by smeubank** [comment]
> I believe this is still blocked by https://getsentry.atlassian.net/browse/INGEST-815 right?

i can't open this ticekt, does it still exist? 

Is the API surface around attachements even valuable enough to keep this around?

**Comment by philipphofmann** [comment]
We didn't do this in the last two years. Let's close it and open a new issue if required.
